### PR TITLE
fix communication port duplicate issue

### PR DESCRIPTION
**Pull request by hjliu0206**
select comm_port and select kernel 5 port maybe duplicated

### CODE REVIEW

**Comment by hjliu0206** [comment]
i raised an request to  fix the issue 
https://github.com/jupyter-server/gateway_provisioners/issues/133

File: gateway_provisioners/kernel-launchers/shared/scripts/server_listener.py [hunk 1]
```diff
@@ -61,7 +61,7 @@ def __init__(
         self.comm_socket: socket | None = None
 
     def build_connection_file(self) -> None:
```
Comment (kevin-bates) [inline gateway_provisioners/kernel-launchers/shared/scripts/server_listener.py:63]: Could you please update the signature to indicate that the _communication_ port is returned?

File: gateway_provisioners/kernel-launchers/shared/scripts/server_listener.py [hunk 2]
```diff
@@ -98,7 +99,7 @@ def _encrypt(self, connection_info_bytes: bytes) -> bytes:
         b64_payload = base64.b64encode(json.dumps(payload).encode(encoding="utf-8"))
         return b64_payload
 
-    def return_connection_info(self) -> None:
+    def return_connection_info(self, common_port) -> None:
```
Comment (kevin-bates) [inline gateway_provisioners/kernel-launchers/shared/scripts/server_listener.py:102]: The current `comm_port` represents _communication_ port, not _common_ port.  I'm fine with either `communication_port` or `comm_port` as the variable name (here and throughout this file)

**Review by kevin-bates — COMMENTED** [review]
Hi @hjliu0206 - thank you for the issue and PR - this is greatly appreciated and a great catch!

I just had a couple comments on naming and signatures.

File: gateway_provisioners/kernel-launchers/shared/scripts/server_listener.py (hunk 1 above)
Comment (hjliu0206) [reply gateway_provisioners/kernel-launchers/shared/scripts/server_listener.py:63]: hello, i already changed the variable name

File: gateway_provisioners/kernel-launchers/shared/scripts/server_listener.py (hunk 2 above)
Comment (hjliu0206) [reply gateway_provisioners/kernel-launchers/shared/scripts/server_listener.py]: hello, i already changed the communication_port name , could you help a review, and merge my code ?

File: gateway_provisioners/kernel-launchers/shared/scripts/server_listener.py (hunk 1 above)
Comment (kevin-bates) [inline gateway_provisioners/kernel-launchers/shared/scripts/server_listener.py:63]: This signature should be updated....
```suggestion
    def build_connection_file(self) -> int:
```
Please rename `com_port` to `comm_port` for consistency.

**Review by kevin-bates — COMMENTED** [review]
We already use `comm_port` in other locations.  Let's consistently use `comm_port` throughout. Thanks.

File: gateway_provisioners/kernel-launchers/shared/scripts/server_listener.py (hunk 1 above)
Comment (hjliu0206) [reply gateway_provisioners/kernel-launchers/shared/scripts/server_listener.py]: thanks , already update the signature and the variable name

**Review by kevin-bates — APPROVED** [review]
LGTM - thank you @hjliu0206

**Pull request merged**